lib/spack/spack/modules.py

```diff
@@ -430,7 +433,7 @@ class Dotkit(EnvModule):
 
     prerequisite_format = None  # TODO : does something like prerequisite exist for dotkit?
 
-    default_naming_format = '{name}-{version}-{compiler.name}-{compiler.version}-{hash}'
+    default_naming_format = '{name}-{version}-{compiler.name}-{compiler.version}'
 
```

Comment:
What about adding `self.spec.variants` to the tokens and allowing that to replace the hash? It is not perfect and maybe you have a better idea for making the name more readable.